src/app/specialties/specialty-add/specialty-add.component.html

```diff
@@ -5,22 +5,19 @@ <h2>New Specialty</h2>
       <div class="form-group" hidden="true">
         <input type="text" hidden="true" class="form-control" id="id" [(ngModel)]="speciality.id" name="id" />
       </div>
-      <div *ngIf="
-          specialityName.invalid &&
-          (specialityName.dirty || specialityName.touched)
-        " class="alert alert-danger">
-        Loading...
-        <div *ngIf="specialityName.errors?.required">Name is required.</div>
-      </div>
 
-      <div class="form-group has-feedback">
+      <div class="form-group has-feedback"   [class.has-success]="specialityName.dirty && specialityName.valid" [class.has-error]="specialityName.dirty &&  !specialityName.valid">
         <div class="form-group">
           <label class="col-sm-1 control-label">Name</label>
           <div class="col-sm-6">
-            <input id="name" name="name" class="form-control" type="text" [(ngModel)]="speciality.name" #specialityName="ngModel" />
+            <input id="name" name="name" class="form-control"  minlength="1" maxlength="80" pattern="^[A-Za-z0-9].{0,79}$" required type="text" [(ngModel)]="speciality.name" #specialityName="ngModel" />
```

Comment:
suggestion: in the `openapi.yaml` specification, the `specialty name` doesn't have any pattern:
https://github.com/spring-petclinic/spring-petclinic-rest/blob/master/src/main/resources/openapi.yml#L1909
Earch `String` field should have a pattern.